pkg/utils/kube/ctl.go

```diff
@@ -597,19 +596,6 @@ func (k *KubectlCmd) authReconcile(ctx context.Context, config *rest.Config, kub
 	return strings.Join(out, ". "), nil
 }
 
-// ConvertToVersion converts an unstructured object into the specified group/version
-func (k *KubectlCmd) ConvertToVersion(obj *unstructured.Unstructured, group string, version string) (*unstructured.Unstructured, error) {
```

Comment:
GitOps engine "watch" preferred versions of the resources to minimize memory usage. So we have to rely on conversion to avoid making an API cal and do client-side only diffing. It is critical for performance when the engine is used to manage multiple clusters.